R/lint.R

```diff
@@ -182,20 +182,11 @@ lint_dir <- function(path = ".", ...,
     return(lints)
   }
 
-  pb <- if (isTRUE(show_progress)) {
-    txtProgressBar(max = length(files), style = 3L)
-  }
-
   lints <- flatten_lints(lapply(
-    files,
-    function(file) {
-      maybe_report_progress(pb)
-      lint(file, ..., parse_settings = FALSE, exclusions = exclusions)
-    }
+    cli::cli_progress_along(files, name = "Running linters"),
+    function(idx) lint(files[idx], ..., parse_settings = FALSE, exclusions = exclusions)
```

Comment:
cf. https://cli.r-lib.org/articles/progress.html#progress-bars-for-mapping-functions-cli_progress_along

![Uploading Screenshot 2024-08-03 at 16.44.40.png…]()